tinygrad/codegen/uopgraph.py

```diff
@@ -396,11 +396,12 @@ def do_expand(root:UOp):
   return UOp(UOps.EXPAND, root.dtype, tuple(new_srcs), expand_args)
 
 acc_number = 0
-def do_reduce(root):
+def do_reduce(root:UOp):
   global acc_number
-  reduce_parented, reduce_unparented = partition(root.src[1:], lambda x: x in root.src[0].parents)
+  reduce_parented, reduce_unparented = partition(list(root.src[1:]), lambda x: x in root.src[0].parents)
```

Comment:
change partition function to Iterable instead of change the user code

Reply:
done